File: Data Structures & Algorithms/is-palindrome/submission-3.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int cur_begin = 0, cur_end = s.size() - 1;
        while(cur_begin < cur_end){
            while(cur_begin < cur_end &&!isalnum(s[cur_begin])) cur_begin++;
            while(cur_begin < cur_end &&!isalnum(s[cur_end])) cur_end--;
            if(cur_begin > cur_end) return true;
            if(tolower(s[cur_begin]) == tolower(s[cur_end])){
                cur_begin++;
                cur_end--;
            }
            else return false;
        }
        return true;
    }
};

File: Data Structures & Algorithms/meeting-rooms-iii/submission-3.py
class Solution:
    def mostBooked(self, n: int, meetings: List[List[int]]) -> int:
        roomStatus = [-1] * n
        roomUsage = [0] * n
        meetings = sorted(meetings)
        for m in meetings:
            curBegin, curEnd = m
            #print(curBegin, curEnd)
            assigned = False
            earliestAvailable = [-1, math.inf]
            for i in range(n):
                if roomStatus[i] < earliestAvailable[1]:
                    earliestAvailable = [i, roomStatus[i]]
                if roomStatus[i] <= curBegin:
                    roomStatus[i] = curEnd
                    roomUsage[i] += 1
                    assigned = True
                    break
            if not assigned:
                #print("E", earliestAvailable)
                roomStatus[earliestAvailable[0]] = roomStatus[earliestAvailable[0]] + curEnd - curBegin
                roomUsage[earliestAvailable[0]] += 1
            #print(roomStatus)
        maxRoomUsage = -1
        maxUsedRoom = -1
        for i,usage in enumerate(roomUsage):
            if usage > maxRoomUsage:
                maxUsedRoom = i
                maxRoomUsage = usage
        return maxUsedRoom
        
                
        